### PR TITLE
add ignore events command line arg

### PR DESCRIPTION
**Pull request by cctdaniel**
not the most flexible/granular way -- open to alternatives/suggestions

### CODE REVIEW

File: observer.py [hunk 1]
```diff
@@ -34,7 +34,8 @@ def get_publishers(network):
 
 
 async def main(args):
-    symbols_to_ignore = set(args.ignore) if args.ignore else set()
+    symbols_to_ignore = set(args.ignore_symbols) if args.ignore_symbols else set()
+    events_to_ignore = set(args.ignore_events) if args.ignore_events else set()
```
Comment (jayantk) [inline observer.py:38]: I think this parameterization isn't flexible enough. We have the following use cases:
1) suppress all alerts for a symbol (e.g., ORCA)
2) suppress a specific error code for a symbol (offline for FX pairs)

One somewhat hacky way to do this is to take a list of regexes. Each regex gets matched against the symbol name + "/" + error code. Then you can specify like `Crypto.ORCA.*` to block all orca alerts, or `FX.*/publisher-offline.*` for the second use case.

**Comment by cctdaniel** [comment]
@erancx are we able to modify the github action to use git sha as tag instead of pr?

File: observer.py [hunk 2]
```diff
@@ -75,64 +87,67 @@ async def run_alerts():
...
+                            publishers=publishers,
+                        )
+                        price_account_errors = validators[symbol].verify_price_account(
+                            price_account=price_account,
+                            coingecko_price=coingecko_price,
+                        )
+                        if price_account_errors:
+                            filtered_errors = filter_errors(
+                                args.ignore, price_account_errors) if args.ignore else price_account_errors
+                            errors.extend(filtered_errors)
```
Comment (jayantk) [inline observer.py:117]: nitpick: this filtering logic could be written in a single location if you saved `errors` here and then filtered around line 145 before calling notify.

File: observer.py [hunk 3]
```diff
@@ -33,8 +34,19 @@ def get_publishers(network):
     return json_data.get(network, {})
 
 
+def filter_errors(regexes, errors):
+    filtered_errors = []
+    for e in errors:
+        skip = False
+        for r in regexes:
+            if re.search(r, f"{e.symbol}/{e.error_code}"):
```
Comment (jayantk) [inline observer.py:42]: I think you might want to use `re.match` here and force the user to put `.*` in the regex where they expect wildcard matches. I think that's a little more intuitive than having them put in `^` and `$` to anchor the string.

Also, maybe this should be a case-insensitive match ?

File: observer.py [hunk 4]
```diff
@@ -207,10 +222,9 @@ async def run_coingecko_get_price():
         help="Prometheus Exporter port",
     )
     parser.add_argument(
-        "-i",
         "--ignore",
         nargs="+",
-        help="List of symbols to ignore",
+        help="List of symbols and/or events to ignore",
```
Comment (jayantk) [inline observer.py:227]: Explain how the argument works. Give some examples like `Crypto.BTCUSD/...` so it's clear to the user. E.g., `Crypto.BTCUSD/.*` matches all alerts on BTCUSD.

File: pyth_observer/prices.py [hunk 5]
```diff
@@ -113,6 +113,14 @@ def update_slot(self, slot: Optional[int]) -> None:
         if self.last_updated_slot is None or slot > self.last_updated_slot:
             self.last_updated_slot = slot
 
+    def update_coingecko_price(self, coingecko_price: Optional[float]) -> None:
```
Comment (jayantk) [inline pyth_observer/prices.py:116]: this seems like it's not related to the rest of the change (?) if so, please remove from this PR.

**Review by jayantk — APPROVED** [review]

File: observer.py (hunk 2 above)
Comment (cctdaniel) [reply observer.py:117]: ah you're right, that's way cleaner
